### ISSUE TITLE
cuDNN's runtime-compiled engines (fused attention) can never build a plan: NVRTC/builtins version mix in the bundle

### ISSUE THREAD

**Issue by AsafManela** [EnzymeAD/Reactant.jl#3226]
## Summary

cuDNN's runtime-compiled engines -- fused attention / SDPA and runtime fusion, i.e. the
`__cudnn$fmhaSoftmax` custom call and anything the `--xla_gpu_enable_cudnn_fmha` rewriter
would produce -- can never build an execution plan through `Reactant_jll`. Every plan build
fails in NVRTC:

```
CUDNN_STATUS_INTERNAL_ERROR_COMPILATION_FAILED; Reason: Encountered runtime kernel
compilation failure at: compilationResult != NVRTC_SUCCESS
[cudnn_frontend] Error: No valid execution plans built.
```

The cause is an NVRTC/builtins version mix inside the static link, and there is a one-line
environment hook that works around it. Details below; PR to follow.

## Reproducer

A hand-emitted `stablehlo.custom_call @__cudnn$fmhaSoftmax` with JAX's JSON backend config
(the op `jax.nn.dot_product_attention(implementation="cudnn")` lowers to): BNTH layout,
bf16, causal, B=2, H=6, T=S=512, D=128, 3-D f32 softmax stats. XLA:GPU accepts it -- the
config parses, the cuDNN graph is built, the descriptors validate -- and then the plan build
fails as above. A self-contained version is in the PR as `test/integration/cudnn.jl`.

## What actually fails

- `libReactantExtra.so` statically links NVRTC *and* its builtins (`libnvrtc_static_*`
  symbols, `nvrtc-builtins.compute_{75..121}.c`), so this is not a missing `libnvrtc`.
- `strace -f -e openat` on the process: no `libnvrtc*` / `libnvptxcompiler*` is ever looked
  up. Only XLA's own `ptxas` children run. `LD_LIBRARY_PATH`, `CUDA_PATH` and `CUDA_HOME`
  cannot reach this code path at all.
- cuDNN 9.21's own error log (`CUDNN_LOGLEVEL_DBG=2`) puts the failure inside
  `rtk(kernelNumRunning)->compile(compilerFlags, this->useNvrtcSassPath, true)` -- the SDPA
  engine compiles straight to SASS -- but does not surface the NVRTC diagnostic.
- With gdb attached: cuDNN's `nvrtcHelper::load()` (libcudnn_graph) fills a function-pointer
  table for the NVRTC API. Absent an override it does **not** point at the bundle's public
  `nvrtc*` symbols (libnvrtc_static, `nvrtcVersion` -> 13.1) but at an NVRTC embedded in
  `libcudnn_graph_static` (renamed `libcudnn_graph_static_v9_<sha>` symbols; its
  `nvrtcVersion` -> 13.2, with its own `nvrtcGetErrorString` table).
- Breaking on that embedded `nvrtcCompileProgram` and calling the embedded
  `nvrtcGetProgramLog`:

  ```
  NVRTC_ERROR_INVALID_OPTION
  nvrtc: error: invalid value for --gpu-architecture (-arch)
  ```

  The bundle carries only CUDA 13.1's builtins (`__nvrtc_builtins_static__13_1_*`), i.e. a
  cuDNN-9.21-for-CUDA-13.2 / CUDA-13.1-NVRTC mix in one static link, and the embedded
  compiler cannot resolve the H100 architecture it is handed.

## The hook

`nvrtcHelper::load()` starts with `getenv("CUDNN_NVRTC_OVERRIDE_PATH")`. If it is set, cuDNN
dlopens that library and dlsyms `nvrtcCreateProgram`, `nvrtcCompileProgram`, ... out of it
(failing with `"' not found in NVRTC override library: '"`), checking only that the NVRTC
major version matches CUDART's. Pointing it at a self-consistent shared libnvrtc makes the
SDPA engine compile. CUDA.jl's `libnvrtc.so.13.0.88` and `.13.3.33` both work: the plan
builds, and the output matches a plain-attention reference (max abs diff 7.8e-3, bf16).

Worth having beyond correctness -- at B=2, H=6, T=S=512, D=128, bf16, causal, the cuDNN
fused forward runs in **0.43-0.49 ms** against **4.2-4.4 ms** for XLA's fused plain
attention on the same H100.

## Proposed fix

PR to follow: `Reactant.XLA.set_cudnn_nvrtc_override!(libnvrtc; libnvrtc_builtins, force)`,
which validates a candidate (exists, loads, `nvrtcVersion` major == `Reactant_jll`'s CUDA
major), dlopens the builtins so the soname lookup needs no `LD_LIBRARY_PATH` entry, and
never overwrites a value the user set. `ReactantCUDAExt.__init__` calls it with
`CUDA_Runtime_jll`'s `libnvrtc_path` / `libnvrtc_builtins_path`, so `using CUDA` alongside
Reactant is enough.

The real fix is on the bundle side: link cuDNN's runtime-compiled engines against a matching
NVRTC, or ship `libnvrtc.so` + builtins under `lib/cuda` in `Reactant_jll` and set the
override from there. Happy to redo it that way if you'd rather -- the Julia-side helper is
what I could verify without rebuilding the jll.

Note that this also explains why the `--xla_gpu_enable_cudnn_fmha` rewriter path could never
have produced a working kernel in this bundle. (In the XLA that ships here the auto rewriter
itself appears to be gone, only the flag string remains, so the `__cudnn$fmhaSoftmax` /
`...Backward` custom calls are the way to reach it.) See also EnzymeAD/Enzyme-JAX#994.

## Environment

Reactant.jl main (v0.2.283 source), `Reactant_jll` v0.0.405, CUDA.jl v6.2.2, Julia 1.12.7,
Linux x86_64. H100 80GB, driver 580.105.8 / CUDA 13.0, XLA runtime 13.1, cuDNN 9.21.0.

Verified on that machine after the change: `test/core/cudnn_nvrtc.jl` 12/12,
`test/integration/cudnn.jl` 4/4 (red before it, with "No valid execution plans built").
